### ISSUE TITLE
Endpoint security tests can fail with Fleet resource already exists exception

### ISSUE THREAD

**Issue by cmacknz** [elastic/elastic-agent#3244]
## Flaky Test

* **Test Name:** TestInstallAndCLIUninstallWithEndpointSecurity
* **Branch:** 8.9
* **Artifact Link:** https://buildkite.com/elastic/elastic-agent/builds/2487

### Stack Trace

```
=== RUN   TestInstallAndCLIUninstallWithEndpointSecurity
    endpoint_security_test.go:68: Enrolling the agent in Fleet
    endpoint_security_test.go:79: Creating enrollment API key...
    endpoint_security_test.go:79: Unpacking and installing Elastic Agent
    fetcher.go:88: Using existing artifact elastic-agent-8.9.1-SNAPSHOT-linux-arm64.tar.gz
    fixture.go:211: Extracting artifact elastic-agent-8.9.1-SNAPSHOT-linux-arm64.tar.gz to /tmp/TestInstallAndCLIUninstallWithEndpointSecurity3653345831/001
    fixture.go:224: Completed extraction of artifact elastic-agent-8.9.1-SNAPSHOT-linux-arm64.tar.gz to /tmp/TestInstallAndCLIUninstallWithEndpointSecurity3653345831/001
    fixture.go:432: Components were not modified from the fetched artifact
    fixture.go:358: >> running agent with: [/tmp/TestInstallAndCLIUninstallWithEndpointSecurity3653345831/001/elastic-agent-8.9.1-SNAPSHOT-linux-arm64/elastic-agent install --force --non-interactive --url https://4f5bd79f886a46c48131adbbd570e673.fleet.us-central1.gcp.qa.cld.elstc.co:443 --enrollment-token RmY3QTk0a0JjSklxdnJJeHFxNXU6enUxcm1udlJTX1MteC1MQXNUNkpuZw==]
    endpoint_security_test.go:79: >>> Ran Enroll. Output: Installing in non-interactive mode.{"log.level":"info","@timestamp":"2023-08-15T05:52:55.407Z","log.origin":{"file.name":"cmd/enroll_cmd.go","file.line":478},"message":"Starting enrollment to URL: https://4f5bd79f886a46c48131adbbd570e673.fleet.us-central1.gcp.qa.cld.elstc.co:443/","ecs.version":"1.6.0"}
        {"log.level":"info","@timestamp":"2023-08-15T05:52:56.735Z","log.origin":{"file.name":"cmd/enroll_cmd.go","file.line":276},"message":"Successfully triggered restart on running Elastic Agent.","ecs.version":"1.6.0"}
        Successfully enrolled the Elastic Agent.
        Elastic Agent has been successfully installed.
    tools.go:34: Agent status: enrolling
    endpoint_security_test.go:81: Installing Elastic Defend
    endpoint_security_test.go:82: Templating endpoint package policy request
    endpoint_security_test.go:82: POST /api/fleet/package_policies
    endpoint_security_test.go:82: 
        	Error Trace:	/home/ubuntu/agent/testing/integration/endpoint_security_test.go:246
        	            				/home/ubuntu/agent/testing/integration/endpoint_security_test.go:82
        	Error:      	Received unexpected error:
        	            	http error response with code 400: {StatusCode:400 Error:Bad Request Message:Error installing endpoint 8.9.0: resource_already_exists_exception
        	            		Root causes:
        	            			resource_already_exists_exception: index [.fleet-files-endpoint-000001/CSX60CW_T1eSp2iBUnND-A] already exists}
        	Test:       	TestInstallAndCLIUninstallWithEndpointSecurity
    fixture_install.go:116: collecting diagnostics; test failed
    fixture.go:358: >> running agent with: [/opt/Elastic/Agent/elastic-agent diagnostics -f /home/ubuntu/agent/build/diagnostics/TestInstallAndCLIUninstallWithEndpointSecurity-diagnostics-2023-08-15T05:53:43Z.zip]
    fixture.go:358: >> running agent with: [/opt/Elastic/Agent/elastic-agent uninstall --force]
--- FAIL: TestInstallAndCLIUninstallWithEndpointSecurity (83.76s)
```

**Comment by cmacknz** [comment]
If this doesn't happen on main we can likely close it, there were fixes on the Fleet side for problems like this.

**Comment by cmacknz** [comment]
The fix I was thinking of should be in 8.9 https://github.com/elastic/kibana/pull/162724

**Comment by pierrehilbert** [comment]
Seems to not happening anymore. Will close for now.
